### PR TITLE
Homework/print cargo

### PR DESCRIPTION
**Pull request by kuchichan**
Renaming to lower case files
print cargo
rebasing and removing unnecessary things
erase warning with char* constexpr

### CODE REVIEW

File: shm/src/print_cargo.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#include "print_cargo.hpp"
+
+void PrintCargo::execute(Player* player) {
+    size_t cargoIndex = 0;
+    std::string dash(42, '-');
+    auto cargo = player->getCargo(cargoIndex);
+    
+    std::cout << std::setw(14) << "Name |" << std::setw(14) << " Amount |"
+              << std::setw(15) << " Baseprice |\n"
+              << dash << '\n';
```
Comment (WojtekMs) [inline shm/src/print_cargo.cpp:17]: Lines 9 + 12 - 14 I would put into another f-ction called for example: displayCargoHeaders()

File: shm/src/print_cargo.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    while (cargo) {
+        std::cout << std::setw(12) << cargo->getName() << " |" << std::setw(12)
+                  << cargo->getAmount() << " |" << std::setw(12) << cargo->getBasePrice()
+                  << " |\n";
```
Comment (WojtekMs) [inline shm/src/print_cargo.cpp]: I would put this into another f-ction called for example: displayCargoValues(cargo)

Comment (WojtekMs) [inline shm/src/print_cargo.cpp:19]: I would put this into another f-ction called for example: displayCargoValues()

File: shm/src/print_cargo.cpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#include <iomanip>
+#include <iostream>
+
+#include "player.hpp"
+#include "print_cargo.hpp"
+
+void PrintCargo::execute(Player* player) {
+    size_t cargoIndex = 0;
```
Comment (WojtekMs) [inline shm/src/print_cargo.cpp]: I like this name, it's descriptive 👍

File: shm/tests/NextDayTest.cpp [hunk 4]
```diff
@@ -67,7 +67,6 @@ TEST_F(NextDayTest, NextDayShouldChangeStockInStore) {
 
 TEST_F(NextDayTest, NextDayShouldPayCrew) {
     auto player_money = player.getMoney();
-    Ship ship = Ship(30, 10, 1, &player);
```
Comment (WojtekMs) [inline shm/tests/NextDayTest.cpp:70]: why did you remove this line?

File: shm/include/map.hpp [hunk 5]
```diff
@@ -12,7 +12,6 @@ class Map {
     const std::vector<Island>& getAllIslands() { return Islands_; };
     void travel(Island* destination);
     size_t getDistanceToIsland(Island* destination);
-    Island* getCurrentPosition() { return currentPosition_; };
```
Comment (WojtekMs) [inline shm/include/map.hpp:15]: Why did you remove this line?

File: shm/src/ship.cpp [hunk 6]
```diff
@@ -19,11 +19,11 @@ Ship::Ship(int capacity,
       speed_(speed),
       name_(name),
       id_(id) {
-  TimeServiceLocator::getTime()->addObserver(this);
+    TimeServiceLocator::getTime()->addObserver(this);
 }
 
 Ship::Ship(int maxCrew, int speed, size_t id, Delegate* delegate)
-    : Ship(0, maxCrew, speed, "", id, delegate) {}
+    : Ship(100, maxCrew, speed, "", id, delegate) {}
```
Comment (WojtekMs) [inline shm/src/ship.cpp:26]: I think about removing this entire c-tor...

**Review by WojtekMs — CHANGES_REQUESTED** [review]
Please change the removed line from island.hpp and NextDayTest

File: shm/tests/NextDayTest.cpp (hunk 4 above)
Comment (kuchichan) [reply shm/tests/NextDayTest.cpp:70]: Accidentally - but please look that we already have identical `Ship` in fixture - ship(30, 10, 1, &player) {}, so it's event better ;d

File: shm/include/map.hpp (hunk 5 above)
Comment (kuchichan) [reply shm/include/map.hpp:15]: Line 11: `Island* getCurrentPosition() const { return currentPosition_; };` <--- was duplicated, this version is even better (has const - preferred in getters). Great read -> https://stackoverflow.com/questions/21478342/c-const-in-getter

File: shm/src/ship.cpp (hunk 6 above)
Comment (kuchichan) [reply shm/src/ship.cpp:26]: We will remove that, however AFAIK it was a part of some homework oop1 - so i will leave it until official PR.

File: shm/tests/NextDayTest.cpp (hunk 4 above)
Comment (WojtekMs) [reply shm/tests/NextDayTest.cpp:70]: You are right

File: shm/include/map.hpp (hunk 5 above)
Comment (WojtekMs) [reply shm/include/map.hpp:15]: Yes I agree about the const when it comes to getters. I thought that I removed this second getter without const but I didn't. No more questions :)

**Review by WojtekMs — APPROVED** [review]

File: shm/include/map.hpp (hunk 5 above)
Comment (kuchichan) [reply shm/include/map.hpp:15]: No problem - maybe some mess was created during merging or smth like that.
